core/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -139,4 +140,14 @@
      */
     boolean readOnly();
 
+    /**
+     * Resolves snapshot name(s) in the repository to {@link SnapshotId}(s).  If any snapshots
+     * are not found, a {@link SnapshotMissingException} is thrown.
+     *
+     * @param snapshotNames snapshot name(s)
+     * @return snapshot ids
+     * @throws SnapshotMissingException if a snapshot is not found
+     */
+    List<SnapshotId> resolveSnapshots(String... snapshotNames);
```

Comment:
I struggled with this one myself, the reason I used `String...` is because it makes the API easier for when we only want to resolve a single snapshot name.  But it is quite trivial to create a list out of a single name, so I will go with what you suggested for the sake of consistency.